core/arch/instr.h

```diff
@@ -2489,6 +2526,16 @@ DR_API
 instr_t *
 instr_create_1dst_1src(dcontext_t *dcontext, int opcode, opnd_t dst, opnd_t src);
 
+DR_API
+/**
+ * Convenience routine that returns an initialized instr_t allocated on the
+ * thread-local heap with opcode \p opcode, one destination (\p dst), and
+ * one source (\p src). A hint is set to use the x86 evex form if available
+ * when encoding the instruction.
+ */
+instr_t *
+instr_create_1dst_1src_evex(dcontext_t *dcontext, int opcode, opnd_t dst, opnd_t src);
```

Comment:
Instead of having multiple specialized create_Ndst_Nsrc routines, I would have instr_set_encoding_hint() return the instr_t* back, so we can chain it, just we do for instr_set_predicate and instr_set_translation.  You could make a macro shortcut in instr_create_shared.h like those 2 have if desired.

Reply:
Done.